src/components/KRadio/KRadio.vue

```diff
@@ -1,54 +1,72 @@
 <template>
   <div
-    :checked="isSelected"
     class="k-radio"
     :class="[
-      isTypeDefault ? 'k-radio-default' : `k-radio-${type}`,
       $attrs.class ? $attrs.class : '',
-      { 'disabled': isDisabled }
+      { 'disabled': isDisabled, 'k-radio-card': isCard, 'has-error': hasError, 'checked': isChecked }
     ]"
   >
     <input
       :id="inputId"
-      :checked="isSelected"
+      :checked="isChecked"
       v-bind="modifiedAttrs"
       class="k-input"
       :disabled="isDisabled"
       type="radio"
       @click="handleClick"
     >
 
-    <KLabel
-      v-if="isTypeDefault && hasLabel"
-      v-bind="labelAttributes"
-      class="k-radio-label"
+    <div
+      v-if="!isCard && (label || $slots.default)"
+      class="radio-label-wrapper"
       :class="{ 'has-description': hasDescription }"
-      :for="inputId"
     >
-      <slot>{{ label }}</slot>
+      <KLabel
+        v-bind="labelAttributes"
+        class="radio-label"
+        :for="inputId"
+      >
+        <slot>{{ label }}</slot>
+
+        <template
+          v-if="hasTooltip"
+          #tooltip
+        >
+          <slot name="tooltip" />
+        </template>
+      </KLabel>
 
-      <div
+      <p
         v-if="hasDescription"
-        class="k-radio-description"
+        class="radio-description"
       >
         <slot name="description">
           {{ description }}
         </slot>
-      </div>
-
-      <template
-        v-if="hasTooltip"
-        #tooltip
-      >
-        <slot name="tooltip" />
-      </template>
-    </KLabel>
+      </p>
+    </div>
 
     <label
-      v-else-if="$slots.default"
+      v-else-if="label || $slots.default"
+      class="card-radio-wrapper radio-label-wrapper"
       :for="inputId"
+      :tabindex="isDisabled ? -1 : 0"
     >
-      <slot name="default" />
+      <slot />
+      <span
+        v-if="label"
+        class="radio-label"
+      >
+        {{ label }}
+      </span>
+      <p
+        v-if="hasDescription"
+        class="radio-description"
+      >
+        <slot name="description">
+          {{ description }}
+        </slot>
+      </p>
```

Comment:
question: instead of using a `p` tag here, shouldn't the description be contained within the `label` so it's clickable?

Reply:
Then my point still stands -- a `p` tag is also not valid HTML then since the user can slot in `div` etc.

Reply:
Fair enough - updated